ten-simple-rules-dockerfiles.Rmd

```diff
@@ -684,10 +697,10 @@ Online collaboration platforms (e.g., GitHub, GitLab) also make it easy to use C
 Continuous integration increases stability and trust, and gives the ability to publish images automatically.
 Automation strategies exist to build and test images for multiple platforms and software versions, even with CI.
 Such approaches are often used when developing popular software packages for a broad user base operating across a wide range of target platforms and environments, and they can be leveraged if you expect your workflow to fall into this category.
-Furthermore, the commit messages in your version controlled repository preserve a record of all changes to the `Dockerfile`.
+Furthermore, the commit messages in your version controlled repository preserve a record of all changes to the `Dockerfile`, and you can use the same versions in tags for both the container image and the git repository.
 
 While there are exceptions to the rule (cf.&nbsp;@kim_bio-docklets_2017), it is generally a simple and clear approach to provide one `Dockerfile` per project.
-Alternatively, you could build an _image stack_ to serve different use cases with a common base image, but this reduces understandability and scatters information across multiple places.
+Alternatively, you could build, or even better contribute to existing, _image stacks_ to serve different use cases with a common base image, but this reduces understandability and scatters information across multiple places.
```

Comment:
Can you explain this more a bit for me? Why would using a popular image stack scatter information ? Don't we advise to use well known / developed bases and this contradicts that?

Reply:
The idea is to say "before you write your own stack, see if you can contribute to an existing (community-maintained?) one".

Reply:
But that might already be covered in rule 1, which could be a better place for that piece of advice (contribute > build own), IF we choose to leave it in.

Reply:
Ohh I see - I think we should rewrite this then, because I read it as "If you use a stack you will reduce understandability and scatter information across multiple places."

Reply:
Well, that's actually what I think. If you (without good reason) use a stack based on 3 or 4 Dockerfiles just to keep these short, then you reduce understandability.

Reply:
Oh so stack == bootstraps? That’s not clear because normally I think of image sets when someone says stack.

Reply:
Here is some better feedback on why this is confusing - when I hear "stack" I think of a stack of containers, not as in bootstrapped from one another, but a group of services. E.g., https://docs.docker.com/engine/swarm/stack-deploy/ - your docker "stack" is your set of containers that are orchestrated together, not one container that has been bootstrapped many times for a final recipe.

Reply:
Maybe instead here you might refer to multiple multi-stage builds, or bootstraps instead of stack?

Reply:
Ahh, we indeed understand the term differently. I was thinking of a stack of images that are starting `FROM` one another, see https://github.com/jupyter/docker-stacks and https://www.rocker-project.org/images/, not a _set_ of containers running together.

So I tried to clear this up, and also moved the "collaborate instead of running your own" a bit down within the rule.